src/MoBi.Core/Commands/AddPathAndValueEntityFromQuantityInSimulationCommand.cs

```diff
@@ -45,27 +57,79 @@ protected override void ClearReferences()
          base.ClearReferences();
          _buildingBlock = null;
          _quantity = null;
-      }
-
-      protected override ICommand<IMoBiContext> GetInverseCommand(IMoBiContext context)
-      {
-         return new RemovePathAndValueEntityFromBuildingBlockInSimulationCommand<TPathAndValueEntity>(_objectPath, _buildingBlock)
-         {
-            Visible = Visible
-         }.AsInverseFor(this);
+         _simulation = null;
       }
 
       public override void RestoreExecutionData(IMoBiContext context)
       {
          base.RestoreExecutionData(context);
+         _simulation = context.CurrentProject.Simulations.FindById(_simulationId);
          _buildingBlock = context.Get<PathAndValueEntityBuildingBlock<TPathAndValueEntity>>(_buildingBlockId);
+         _quantity = _simulation.All<TQuantity>().FindById(_quantityId);
+      }
+
+      protected abstract class RemovePathAndValueEntityFromBuildingBlockInSimulationCommand<T> : BuildingBlockChangeCommandBase<PathAndValueEntityBuildingBlock<T>> where T : PathAndValueEntity
```

Comment:
These removers are nested and protected/private because you should not normally use them unless as an undo for the add.